### PR TITLE
Build mode light maker

### PR DESCRIPTION
**Pull request by cuddleandtea**
## About The Pull Request
Hello dear comrades! Adds new submode for admins buildmode. You can create lights on any turf/mob/obj with different range, power and color.
based on: #2902 because it's changes the same .dmi file with sprites

## Why It's Good For The Game
Here is how it work in the video: https://youtu.be/Ias2_tdiLeU
Admins can light up fast any areas and points of interest or just dark places that needs to be lit. A very handy tool for events and stuff.
1. Go to build mode and select my little lightbulb icon
image
2. Already you can place light with power 1, range 3 and white color. Press Build Mode button to change this settings. 
image
image
3. Enjoy this great tool for events and points of interest
image

## Changelog

:cl:
admin: new light maker subtype for build mode
/:cl:

### CODE REVIEW

File: code/modules/buildmode/submodes/lightmaker.dm [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		if("Color")
+			light_color = input(target_client, "Light color", text("Input")) as color|null
+
+/datum/buildmode_mode/lightmaker/handle_click(client/target_client, params, obj/object)
+	var/list/modifiers = params2list(params)
+
+	if(LAZYACCESS(modifiers, LEFT_CLICK))
+		object.set_light(light_range, light_power, light_color)
+	if(LAZYACCESS(modifiers, RIGHT_CLICK))
+		object.set_light(0,0,0, COLOR_WHITE)
```
Comment (thgvr) [inline code/modules/buildmode/submodes/lightmaker.dm]: That's not exactly deleting the light source

Comment (cuddleandtea) [reply code/modules/buildmode/submodes/lightmaker.dm]: > That's not exactly deleting the light source

Creating room with no light 
image
VV: 
```
(E) (C) (M) light_color = "#FFFFFF"
(E) (C) (M) light_flags = 0
(E) (C) (M) light_on = 1
(E) (C) (M) light_power = 1
(E) (C) (M) light_range = 0
(E) (C) (M) light_sources = null
(E) (C) (M) light_system = 1
```
1. Deleting light by my code: 
```
(E) (C) (M) light_color = 0
(E) (C) (M) light_flags = 0
(E) (C) (M) light_on = "#FFFFFF" //now I see the error - it was working but still not accurate. 
(E) (C) (M) light_power = 0
(E) (C) (M) light_range = 0
(E) (C) (M) light_sources = null
(E) (C) (M) light_system = 1
```
2. Adding light with my code: 
```
(E) (C) (M) light_color = "#FFFFFF"
(E) (C) (M) light_flags = 0
(E) (C) (M) light_on = "#FFFFFF"
(E) (C) (M) light_power = 1
(E) (C) (M) light_range = 3
(E) (C) (M) light_sources = /list (1)
1 = /datum/light_source [0x210054c1]
(E) (C) (M) light_system = 1
```

**Comment by cuddleandtea** [comment]
@thgvr 
I don't know how to "delete" light by other ways. Technically it's setting light vars so the object doesn't glow. Here is how it work in the video: https://youtu.be/Ias2_tdiLeU

**Comment by cuddleandtea** [comment]
The icons/misc/buildmode.dmi conflict is alright. This pull request includes icon from latest change of this .dmi in #2902 and adds the new sprite for lightmaker.
image

**Comment by thgvr** [comment]
> The icons/misc/buildmode.dmi conflict is alright. This pull request includes icon from latest change of this .dmi in #2902 and adds the new sprite for lightmaker. image

To easily fix this, pull updated master to your working branch, and head over to `tools/dmi/Resolve Icon Conflicts.bat` and run it. It should resolve the icon conflicts for you.

**Comment by cuddleandtea** [comment]
> To easily fix this, pull updated master to your working branch, and head over to `tools/dmi/Resolve Icon Conflicts.bat` and run it. It should resolve the icon conflicts for you.

Wow, I never knew that you can do it like that! Thanks!

**Comment by cuddleandtea** [comment]
Runtime again :c 
```Runtime in code/__HELPERS/unsorted.dm,1127: Value is not a reference ByondValue("Null", "NULL")```

**Comment by thgvr** [comment]
> Runtime again :c `Runtime in code/__HELPERS/unsorted.dm,1127: Value is not a reference ByondValue("Null", "NULL")`

This error is just something we have that needs to be fixed, not your fault.

**Comment by thgvr** [comment]
As I was sleeping last night I had an idea for how to make this better

Instead of editing a turf's lighting value, this should spawn an invisible seen-by-ghosts-only type object (similar to the sound emitter) that can then be edited.

**Comment by cuddleandtea** [comment]
> As I was sleeping last night I had an idea for how to make this better
> 
> Instead of editing a turf's lighting value, this should spawn an invisible seen-by-ghosts-only type object (similar to the sound emitter) that can then be edited.

This is the first idea that came to my mind. But in the process I realized that it doesn't allow to interact with moving obj and mobs. Sure, we can connect obj and mobs, but it will be much slower and automation is harder. 

The current code of this tool allows us to use it on anything (glowing creepy monsters that hide in the distance and as soon as you see a red light you have to run, or glowing crystals that the player can pick up and break the seal of darkness with them). 

This implementation of light manipulation through VV is an incredibly fast and comfortable tool to use. Therefore, I kindly ask that this tool be adopted in this form and shape, as it provides the best range of possibilities. Thanks @thgvr

**Comment by cuddleandtea** [comment]
@thgvr Sorry for reminding, can you approve?

**Comment by cuddleandtea** [comment]
@thgvr hello! Kind reminder about PR. Or I shouldn't ping you?

**Comment by cuddleandtea** [comment]
@thgvr Hello! Kind reminder about PR

**Review by FalloutFalcon — APPROVED** [review]
seems fine as a concept, code looks normal, fills a unique use case.

**Pull request merged**